### PR TITLE
Sign Flow - Security Modals

### PR DESCRIPTION
**Pull request by rocky-fleek**
## Changelog

- Adds new modals to verify update calls on asset canisters.
- Transfers show a nice view of the nft being interacted.
- Non transfers show canister details
- Any request without decoded arguments is flagged as unsafe as we cannot show the user what is being done on its behalf/

### Type of changes included:

- [X] Components
- [X] Business Logic
- [X] Bug fixes
- [X] Styling
- [ ] Code Refactor

### Screenshots/Gifs:
image
image
image
image
image
image
image
image
image
image

### Tested on:

- [X] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Pages/Notification/components/Sign/components/Details/components/DisplayBox.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            onClick={toggleModal}
+            alt="more-info"
+          />
+          )}
+        </h2>
+        <p className={classes.amountDescription}>
+          <span>{subtitle || '???'}&nbsp;</span>
+        </p>
+      </div>
+      <img src={img || request?.canisterIcon} className={classes.assetImg} alt="asset img" />
```
Comment (maksym-boytsov) [inline source/Pages/Notification/components/Sign/components/Details/components/DisplayBox.jsx:32]: no need for `img` text in `alt` - devtools will shout at us :D
just `asset` will be ok

File: source/Pages/Notification/components/Sign/components/Details/components/NFTDisplay.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    if (request?.decodedArguments) {
+      const nftId = getNFTId(request);
+      setTitle(`#${Number.isNaN(Number(nftId)) ? decodeTokenId(nftId) : nftId}`);
+    }
+  }, [request]);
+  return (
+    <DisplayBox
+      shouldWarn={shouldWarn}
+      title={title ?? 'Unknown ID or recipient'}
+      subtitle={request?.canisterName || 'Unknown Collection'}
```
Comment (maksym-boytsov) [inline source/Pages/Notification/components/Sign/components/Details/components/NFTDisplay.jsx:38]: Same as above

File: source/Pages/Notification/components/Sign/components/WarningModal/index.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { Button } from '@ui';
+
+import useStyles from './styles';
+
+const WarningModal = ({ toggleModal }) => {
```
Comment (maksym-boytsov) [inline source/Pages/Notification/components/Sign/components/WarningModal/index.jsx:8]: I think it'd be worth to reuse some MUI built-in things to do modals
For now it's totally fine to have it, but MUI do some extra magic like react portals, accessibility, keyboard clicks and stuff like that

File: source/Pages/Notification/components/Sign/components/Details/components/WarningBox.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (rocky-fleek) [reply source/Pages/Notification/components/Sign/components/Details/components/WarningBox.jsx:1]: will leave it as is for now as I like specific definitions whenever possible. If props scale I can destructure them

File: source/Pages/Notification/components/Sign/components/WarningModal/index.jsx (hunk 3 above)
Comment (rocky-fleek) [reply source/Pages/Notification/components/Sign/components/WarningModal/index.jsx:8]: 🤮 MUI imho lol 
cc @angarita-dev back me up hahahah

Comment (maksym-boytsov) [reply source/Pages/Notification/components/Sign/components/WarningModal/index.jsx:8]: Some translations are missing :)

**Pull request merged**